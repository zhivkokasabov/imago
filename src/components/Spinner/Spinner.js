import React from 'react';

import styles from './Spinner.module.scss';

export default function Spinner({
  show
}) {
  if (!show) {
    return null;
  }

  return (
    <div className={styles.spinner}>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  );
}