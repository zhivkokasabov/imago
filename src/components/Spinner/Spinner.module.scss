.spinner {
  width: 5rem;
  height: 5rem;
  display: inline-block;
  position: relative;
  
  div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 4rem;
    height: 4rem;
    margin: .5rem;
    border: .5rem solid rgb(var(--primary-color));
    border-radius: 50%;
    animation: spinner 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: rgb(var(--primary-color)) transparent transparent transparent;

    &:nth-child(1) {
      animation-delay: -0.45s;
    }

    &:nth-child(2) {
      animation-delay: -0.3s;
    }

    &:nth-child(3) {
      animation-delay: -0.15s;
    }
  }
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
