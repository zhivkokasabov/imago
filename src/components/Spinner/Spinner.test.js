import React from 'react';
import { render } from '@testing-library/react';

import Spinner from './Spinner';

describe('Spinner component', () => {
  it('should render Spinner if show is set to true', () => {
    const { asFragment } = render(<Spinner show />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render Spinner if show is set to false', () => {
    const { asFragment } = render(<Spinner />);

    expect(asFragment()).toMatchSnapshot();
  });
});
