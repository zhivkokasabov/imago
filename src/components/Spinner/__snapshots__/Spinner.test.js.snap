// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should not render Spinner if show is set to false 1`] = `<DocumentFragment />`;

exports[`Spinner component should render Spinner if show is set to true 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
